packages/iov-bns/src/bnsconnection.spec.ts

```diff
@@ -1220,6 +1222,132 @@ describe("BnsConnection", () => {
+        const proposal = (await connection.getProposals()).find(p => p.id === proposalId)!;
+        expect(proposal.votingStartTime).toBeLessThan(Date.now() / 1000);
+        expect(proposal.votingEndTime).toBeGreaterThan(Date.now() / 1000);
+        expect(proposal.state.totalYes).toEqual(0);
+        expect(proposal.state.totalNo).toEqual(0);
+        expect(proposal.state.totalAbstain).toEqual(0);
+        expect(proposal.status).toEqual(ProposalStatus.Submitted);
+        expect(proposal.result).toEqual(ProposalResult.Undefined);
+        expect(proposal.executorResult).toEqual(ProposalExecutorResult.NotRun);
+      }
+
+      {
+        const voteForProposal = await connection.withDefaultFee<VoteTx & WithCreator>({
+          kind: "bns/vote",
+          creator: author,
+          proposalId: proposalId,
+          selection: VoteOption.Yes,
+        });
+        const nonce = await connection.getNonce({ pubkey: author.pubkey });
+        const signed = await profile.signTransaction(voteForProposal, bnsCodec, nonce);
+        const response = await connection.postTx(bnsCodec.bytesToPost(signed));
+        await response.blockInfo.waitFor(info => !isBlockInfoPending(info));
+      }
+
+      await sleep(5_000);
+
+      {
+        // Election voting period ended but not yet tallyed
+        const proposal = (await connection.getProposals()).find(p => p.id === proposalId)!;
+        expect(proposal.votingEndTime).toBeLessThan(Date.now() / 1000);
+        expect(proposal.state.totalYes).toEqual(1);
+        expect(proposal.state.totalNo).toEqual(0);
+        expect(proposal.state.totalAbstain).toEqual(0);
+        expect(proposal.status).toEqual(ProposalStatus.Submitted);
+        expect(proposal.result).toEqual(ProposalResult.Undefined);
+        expect(proposal.executorResult).toEqual(ProposalExecutorResult.NotRun);
+      }
+
+      {
+        const tallyVotes = await connection.withDefaultFee<TallyTx & WithCreator>({
+          kind: "bns/tally",
+          creator: author,
+          proposalId: proposalId,
+        });
+        const nonce = await connection.getNonce({ pubkey: author.pubkey });
+        const signed = await profile.signTransaction(tallyVotes, bnsCodec, nonce);
+        const response = await connection.postTx(bnsCodec.bytesToPost(signed));
+        await response.blockInfo.waitFor(info => !isBlockInfoPending(info));
+      }
+
+      {
+        // Election ended and accepted
+        const proposal = (await connection.getProposals()).find(p => p.id === proposalId)!;
+        expect(proposal.state.totalYes).toEqual(1);
+        expect(proposal.state.totalNo).toEqual(0);
+        expect(proposal.state.totalAbstain).toEqual(0);
+        expect(proposal.status).toEqual(ProposalStatus.Closed);
+        expect(proposal.result).toEqual(ProposalResult.Accepted);
+        expect(proposal.executorResult).toEqual(ProposalExecutorResult.Succeeded);
+      }
+
+      connection.disconnect();
+    }, 30_000);
```

Comment:
Is this enough to pass CI reliably?

Reply:
I had no issues with that so far. Locally the test runs in about 12 seconds if I remember correctly.